### PR TITLE
DOC: docs for using a dotplot to evaluate alignment quality, fixes #1063

### LINKED ISSUE #1063: document using a dotplot to evaluate alignment quality
**Describe the solution you'd like**
The dotplot algorithm can show both matching segments between sequences AND the path an alignment algorithm has chosen.

Write a gallery document that shows this feature, explaining how it's useful.

The document should demonstrate using an aligner with bad parameter values. Apply it to the brca1.fasta file. Embellish the following with explanatory text of alignment parameters, what a dotplot shows, how to modify the dotplot display, etc...
```python
from cogent3 import load_aligned_seqs
from cogent3.app.align import progressive_align

seqs = load_aligned_seqs("/Users/gavin/repos/Cogent3/doc/data/brca1.fasta", moltype="dna").degap()
seqs = seqs.take_seqs(['LeafNose',
 'FalseVamp',
 'RoundEare',
 'Sloth',
 'Anteater',
 'HairyArma'])

# bad idea! make gap extension unlikely
aligner = progressive_align("nucleotide", indel_length=1e-9, indel_rate=1e-2)
aln = aligner(seqs)
aln[2200:2500].dotplot("HairyArma", "RoundEare").show()
aln[2270:2500] # showing poorly aligned segment

# run with aligner defaults
aligner = progressive_align("nucleotide")
aln = aligner(seqs)
aln[2200:2500].dotplot("HairyArma", "RoundEare").show()
aln[2270:2500] # showing poorly aligned segment
```

### PR DESCRIPTION
**Pull request by KatherineCaley**
@GavinHuttley I am not sure if there is any sphynx magic I need to do to put an image from this section in the image gallery?

### CODE REVIEW

**Comment by GavinHuttley** [comment]
@KatherineCaley there is, [see this example](https://github.com/cogent3/cogent3/blob/0728025923ea79e05fa63882b5e0daee8ae40203/doc/draw/aln/plot_aln-dotplot-2.rst?plain=1#L25) for how to write a single image for a drw example that will be added to the gallery.

**Comment by KatherineCaley** [comment]
> @KatherineCaley there is, [see this example](https://github.com/cogent3/cogent3/blob/0728025923ea79e05fa63882b5e0daee8ae40203/doc/draw/aln/plot_aln-dotplot-2.rst?plain=1#L25) for how to write a single image for a drw example that will be added to the gallery.

Thanks! updating now

File: doc/draw/aln/plot_aln-dotplot-3.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+----------------------------------------------------
+
+Let's align the same sequences, but with more biologically realistic parameters. The default parameters for the progressive aligner are a better choice as they have a low probability of opening a gap, but a high probability of extending a gap.
+
+.. note:: The default parameters (of any program) are not always the best choice, but they are a good starting point for many alignments.
+
+.. jupyter-execute::
+    :raises:
+
+    aligner = progressive_align("nucleotide")
```
Comment (GavinHuttley) [inline doc/draw/aln/plot_aln-dotplot-3.rst:65]: change to
```python
aligner = get_app("progressive_align", "nucleotide")
```

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Awesome! See my response to your question about how to add a thumbnail.

**Comment by KatherineCaley** [comment]
@GavinHuttley updated to address comments! 🦖

**Review by GavinHuttley — APPROVED** [review]
Awesome!

**Pull request merged**